test/controllers/EditSetControllerSpec.scala

```diff
@@ -69,846 +66,89 @@ class EditSetControllerSpec extends BaseSpec {
           stub.fileMimeTypes,
           stub.executionContext
         ),
-        editSetService,
-        editSetInstance
+        mockEditSetService,
+        mockEditSet
       )
-      val editSet = controller
-        .view("1")
-        .apply(
-          CSRFTokenHelper.addCSRFToken(
-            FakeRequest(GET, "/edit-set/1").withSession(SessionKeys.token -> validSessionToken)
-          )
+      when(mockEditSetService.getEditSet("1")).thenReturn(getEditSet())
+      when(
+        mockEditSet(any[User], anyString(), anyString(), any[EditSetPage], any[EditSetEntryRowOrder])(
+          any[Messages],
+          any[Request[AnyContent]]
         )
-
-      status(editSet) mustBe OK
-      contentType(editSet) mustBe Some("text/html")
-      val document = asDocument(editSet)
-      document must haveCaption("Showing 1 - 10 of 12 records")
-      document must haveHeader("Edit set: COAL 80 Sample")
-    }
-
-    "render the edit set page from the application" in {
-      val controller = inject[EditSetController]
+      ).thenReturn(getEditSetDisplay)
       val editSet = controller
         .view("1")
         .apply(
           CSRFTokenHelper.addCSRFToken(
-            FakeRequest(GET, "/edit-set/1")
-              .withSession(SessionKeys.token -> validSessionToken)
+            FakeRequest(GET, "/edit-set/1").withSession(SessionKeys.token -> validSessionToken)
           )
         )
 
       status(editSet) mustBe OK
       contentType(editSet) mustBe Some("text/html")
-      val document = asDocument(editSet)
```

Comment:
Why have the content assertions been removed?

Reply:
Since the unit test returns some mock `HTML` data, the assertion is limited to the content type and the status of the request

Reply:
Which begs the question (as we discussed): don't we want to allow Play to generate the actual view?

Otherwise, what are we really testing here?

Reply:
I agree - I'm a bit wary here of mocking out the views here, especially as we've not added anything to the `EditSetViewSpec` - I think it would be useful to test some aspects of the views in isolation without running the integration test suite.

In a way I see the HTML as the output of this controller and I would expect assertions to be made on the filled out template - so we can stub the data that the controller uses to fill out the HTML then make assertions on that.

Reply:
`EditSetViewSpec` has some assertions on the content (more than the `EditSetControllerSpec` but much less than in `EditSetISpec`); perhaps this can be addressed in [the next ticket](https://national-archives.atlassian.net/browse/PACT-793).

Reply:
I understand now that @rwalpole wishes for all content assertion to be one in `EditSetViewSpec`; only the status should be asserted here.

However, I see that we're still asserting the content type. Should we?

Reply:
@tna-erasmos @userjd I don't see any benefit in asserting the content type here as we are not setting it explicity in the controller, but rather it is being inferred by the framework.

Reply:
Removed